ScanShop.Mobile/ScanShop.Mobile/Services/HttpClientService.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.IdentityModel.Tokens.Jwt;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using ScanShop.Mobile.Services;
+using Xamarin.Forms;
+
+[assembly: Dependency(typeof(HttpClientService))]
+
+namespace ScanShop.Mobile.Services
+{
+    public class HttpClientService : IHttpClientService
+    {
+        private readonly HttpClient _httpClient;
+
+        public HttpClientService()
+        {
+            _httpClient = new HttpClient();
+            var configurationService = DependencyService.Get<IConfigurationService>();
+            _httpClient.BaseAddress = configurationService.GetBaseUrlAsync().Result;
+        }
+
+        public void SetBearerToken(string token)
+        {
+            _httpClient.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+        }
+
+        public async Task<HttpResponseMessage> PostAsync<T>(string endpoint, T payload)
```

Comment:
why do you need to create additional method like this when PostAsJsonAsync in IHttpClient exist?

Reply:
Fixed in new commit